flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedStateFunction.java

```diff
@@ -25,13 +25,13 @@
  * <p>This functionality is only available through the
  * {@code BroadcastConnectedStream.process(final KeyedBroadcastProcessFunction function)}.
  */
-public abstract class KeyedStateFunction<K, S extends State> {
+public interface KeyedStateFunction<K, S extends State> {
```

Comment:
Add `@FunctionalInterface` annotation.

Reply:
@dawidwys can I change current usage like : 

```
// valid state value via applyToAllKeys().
			backend.applyToAllKeys(VoidNamespace.INSTANCE, VoidNamespaceSerializer.INSTANCE, listStateDescriptor,
				new KeyedStateFunction<Integer, ListState<String>>() {
					@Override
					public void process(Integer key, ListState<String> state) throws Exception {
						assertEquals("Hello" + key, state.get().iterator().next());
					}
				});

			// clear state via applyToAllKeys().
			backend.applyToAllKeys(VoidNamespace.INSTANCE, VoidNamespaceSerializer.INSTANCE, listStateDescriptor,
				new KeyedStateFunction<Integer, ListState<String>>() {
					@Override
					public void process(Integer key, ListState<String> state) throws Exception {
						state.clear();
					}
				});
```

to lambda function style to verify the change?

Reply:
IMO changing every test that uses this function is not the best practice. Anonymous functions are still a valid option, so there is no need to change valid tests. I would be in favour of adding just a single test that tests lambda can be passed.

Reply:
hi @dawidwys the Travis is green, that means the lambda is right. I think there isn't necessary to  have a extra test? I think changing from anonymous implementation to lambda function can be treat as refactor.

Reply:
Yes, but the refactor is unnecessary and pollutes git history.

Reply:
Yes, and add single test for lambda? Why are you against adding a test for that?